proxy-robots-module/src/main/java/orwell/proxy/robot/RobotElementStateVisitor.java

```diff
@@ -0,0 +1,63 @@
+package orwell.proxy.robot;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import orwell.messages.Robot;
+
+
+/**
+ * Created by Michaël Ludmann on 5/12/15.
+ */
+public class RobotElementStateVisitor implements IRobotElementVisitor {
+    private final static Logger logback = LoggerFactory.getLogger(RobotElementStateVisitor.class);
+    private final Robot.ServerRobotState.Builder serverRobotStateBuilder = Robot.ServerRobotState.newBuilder();
+
+    protected Robot.ServerRobotState getServerRobotState() {
+        return serverRobotStateBuilder.build();
+    }
+
+    /**
+     *
+     * @return the byte array of ServerRobotState or
+     *         null if both rfid and colour list are empty
+     */
+    public byte[] getServerRobotStateBytes() {
+        final Robot.ServerRobotState serverRobotState = getServerRobotState();
+        if (null == serverRobotState ||
+                (serverRobotState.getRfidList().isEmpty() &&
+                serverRobotState.getColourList().isEmpty()))
+            return null;
+        else
+            return serverRobotState.toByteArray();
+    }
+
+    public void reset(){
```

Comment:
We reset the visitor to its initial state.
Will rename it to clearServerRobotState()...